packages/host/app/components/operator-mode/profile-info-popover.gts

```
@@ -192,29 +208,32 @@ export default class ProfileInfoPopover extends Component<ProfileInfoPopoverSign
     return await this.realmServer.fetchCreditInfo();
   });
 
+  private get isLoading() {
+    return this.fetchCreditInfo.isLoading;
+  }
+
   private get plan() {
     return this.fetchCreditInfo.value?.plan;
   }
 
+  private get planName() {
+    return this.plan?.name ?? '-';
```

Probably we don't want a case where user does not have a plan

---

Should I hide the credit section in account popover when user does not subscribe to any plan?

---

No, I mean that I think user should be subscribed to one of the plans in order to use the app. CS-7396 will ensure user starts with a free plan, and then they can switch between plans. I don't see a case here where user is not on a plan

---

I guess the question is what will be shown to the existing users that are already registered and didn't go through the stripe flow to subscribe to the free plan. Do we redirect them to the screen in the mentioned ticket, or maybe add a new button in the popover that takes them to the free plan subscribe payment link, e.g. "Subscribe to our free plan to try using Boxel"? It's something to consider and discuss. It's also confusing for user to see they are not on a plan (`-`) and then there is a button to "Upgrade plan" (upgrading from what?).

---

Does that mean in the endpoint we can throw NotFound error if user has no subscription, instead of returning `{ data: null, included: null }` ?

---

I think it's ok if you return the user data (matrixUserId,..) without subscription data included (or empty subscriptions array)

---

@lukemelia  we’d appreciate your input.

---

A user should not be able to access Boxel operator mode unless they have a plan. Users who do not should be shown a message explaining this with a button to sign up for the free plan.

---

@jurgenwerk, I've updated the code so that the account popover no longer handles the 'no plan' case. I also added a note to this ticket: https://linear.app/cardstack/issue/CS-7396/send-user-to-stripe-checkout-for-free-plan-during-registration-flow#comment-aaba938d.